### PR TITLE
chore: install container w/o dirty repo; run as non-root

### LINKED ISSUE #1743: chore: fix the renku-python version used in the core service
At the moment, because the core service docker container runs a command that modifies the state of the repo, the version of the service is `0.x.y-dirty` - this looks particularly bad now with migrations implemented in the UI - see, for example:

image

There is no need to pollute the repo in the container - we can either not use the requirements builder or we can export the requirements file to a temporary path outside the repo before installing.

### PR DESCRIPTION
**Pull request by rokroskar**
* fixes #1743 by saving the requirements file in a temporary directory instead of in the repository directory.

* refactors the service requirements in setup.py so they are not installed in the CLI

* side benefit - runs the app as a non-privileged user.

### CODE REVIEW

File: Dockerfile.svc [hunk 1]
```diff
@@ -8,15 +8,22 @@ RUN apk add --no-cache --allow-untrusted \
     --repository http://dl-cdn.alpinelinux.org/alpine/latest-stable/main \
     --repository http://nl.alpinelinux.org/alpine/edge/community \
     git-lfs && \
-    git lfs install
+    git lfs install && \
+    addgroup -g 1000 shuhitsu && \
+    adduser -S -u 1000 -G shuhitsu shuhitsu && \
```
Comment (Panaetius) [inline Dockerfile.svc:13]: 🤣 @ shuhitsu

File: Dockerfile.svc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.7-alpine
+FROM python:3.7-alpine as base
```
Comment (Panaetius) [inline Dockerfile.svc:1]: You don't use `base` anywhere, is this needed?

File: Dockerfile.svc (hunk 1 above)
Comment (rokroskar) [reply Dockerfile.svc:13]: our notebook service is kyaku ;)

File: Dockerfile.svc (hunk 2 above)
Comment (rokroskar) [reply Dockerfile.svc]: no, I was trying a few different things and this was left over, thanks!

**Review by Panaetius — DISMISSED** [review]
Thank you 🚀

**Review by Panaetius — DISMISSED** [review]

**Review by Panaetius — APPROVED** [review]

**Pull request merged**